src/Storefront/Framework/Routing/StorefrontResponse.php

```diff
@@ -19,9 +19,12 @@ class StorefrontResponse extends Response
 
     public function getData(): array
     {
-        return $this->data;
+        return $this->data ?? [];
     }
 
+    /**
+     * @deprecated tag:v6.5.0 - parameter `$data` will be typed to `array` strictly
+     */
     public function setData(?array $data): void
     {
         $this->data = $data;
```

Comment:
in addition we can log a deprecation message if `null` is passed to the method

Reply:
You mean this, @mitelg ?

```suggestion
        if (is_null($data)) {
            Feature::triggerDeprecationOrThrow(
                'v6.5.0.0',
                sprintf('Parameter "data" in method "setData" will be strictly typed to "array" in class "%s".', static::class)
            );
        }
        
        $this->data = $data;
```

Reply:
yes, that looks okay for me. but @shopware/ct-core can tell you more :blush:

Reply:
@stephanniewerth yes the deprecation logging looks good :+1: